analyzer/linux/lib/core/packages.py

```diff
@@ -118,8 +119,11 @@ def start(self):
             # Remove the trailing slash (if any)
             self.target = filepath.rstrip("/")
         self.prepare()
-        # self.normal_analysis()
         self.strace_analysis()
+        self.nc.init("logs/strace.log", False)
+        self.thread = Thread(target=self.thread_send_strace_buffer)
+        self.thread.start()
```

Comment:
This thread will start immediately and potetenially miss `self.proc.stderr` when it runs. I think you want to synchroinize its execution with `strace_analysis`. Have a look at [threading event objects](https://docs.python.org/3/library/threading.html#event-objects). You can hold back the thread's execution until self.proc is readable.

And with self.proc.stderr being line-buffered I think - you can probably set a threading event each time a line is ready for reading.